indexes/about-self.js

```diff
@@ -1,72 +1,63 @@
 const bipf = require('bipf')
 const pull = require('pull-stream')
 const pl = require('pull-level')
-const jsonCodec = require('flumecodec/json')
```

Comment:
Careful with this. I'm quite sure this one still needs to be there. Pull-level is really wierd with 'json' and jsonCodec, I remember spending at least an hour tracking down a bug related to that.

Reply:
Sure. What happens is that level-codec accepts either a string or an object. So either `json` or `require('flumecodec/json')`.

https://github.com/Level/codec/blob/fdb2c4fb2db2fe0f51be7d221fd74ee70794a794/index.js#L13-L17

```js
  if (typeof encoding === 'string') encoding = encodings[encoding]
  if (!encoding) encoding = encodings.id
  return encoding
```

The "encodings" above is this: https://github.com/Level/codec/blob/v9.0.2/lib/encodings.js

`flumecodec` is just a thin 1-LOC wrapper around `level-codec`: https://github.com/flumedb/flumecodec/blob/master/json.js IMO we should forget `flumecodec`, it's totally useless.

Note that tests in this PR were not changed, and that we had both `jsonCodec` (imported object) and `'json'` (string) mixed sometimes, and it worked, see: https://github.com/ssb-ngi-pointer/ssb-db2/blob/f7639e9225458f4a342e379a5f4e2384150056db/indexes/about-self.js#L30 and https://github.com/ssb-ngi-pointer/ssb-db2/blob/f7639e9225458f4a342e379a5f4e2384150056db/indexes/about-self.js#L89

So, moving forwards, let's just confidently use `keyEncoding: 'json'` or `keyEncoding: 'utf8'` or `keyEncoding: oneOfThoseStrings` and forget `flumecodec`.

Reply:
Hmm, but there is no test of this close and load again in aboutSelf. Maybe we should add that then?

Reply:
`level-post` does not support string encoding: https://github.com/dominictarr/level-post/blob/9021f153f159aa1d13233908ebe616f650e78f22/index.js#L17